azure-pipelines.yml

```diff
@@ -252,8 +312,14 @@ jobs:
         ISPARENT: True
   steps:
   - bash: |
+      uuid="$(tr -dc '[:alnum:]' </dev/urandom| head -c 16)"
       docker build -t ${DOCKER}:${VERSION}-${VARIANT} --build-arg MAKEFLAGS="-j$(($(grep -c ^processor /proc/cpuinfo) + 1))" docker-images/${VERSION}/${VARIANT}
-      docker run --rm ${DOCKER}:${VERSION}-${VARIANT} -buildconf
+      # Create a uniquely tagged version of the container to avoid pulling a version that was already on docker hub
+      docker tag ${DOCKER}:${VERSION}-${VARIANT} ${DOCKER}:${VERSION}-${VARIANT}-${uuid}
+      # Run -buildconf from the unique version we just tagged to make sure the build was ok
+      docker run --rm ${DOCKER}:${VERSION}-${VARIANT}-${uuid} -buildconf
+      # remove the unnecessary uuid tagged image
+      docker rmi ${DOCKER}:${VERSION}-${VARIANT}-${uuid}
```

Comment:
the whole /var/lib/docker is cleaned after, that shouldn't necessary

Reply:
Will ditch this line then 🙂